src/lib.rs

```diff
@@ -1,6 +1,7 @@
 #![cfg_attr(not(test), no_std)]
 #![feature(impl_trait_projections)]
 #![feature(async_fn_in_trait)]
+#![feature(type_changing_struct_update)]
```

Comment:
How close is this feature to stabilization? I would prefer if it could be avoided.